app/reducers/roles.js

```diff
@@ -0,0 +1,28 @@
+import { ROLES } from '../actions/actionsTypes';
+
+const initialState = {
+	roles: {}
+};
+
+export default function permissions(state = initialState, action) {
+	switch (action.type) {
+		case ROLES.SET:
+			return action.roles;
+		case ROLES.UPDATE:
+			return {
+				...state,
+				[action.payload.id]: action.payload.desc || action.payload.id
+			};
+		case ROLES.ADD:
```

Comment:
`ROLES.UPDATE` and `ROLES.ADD` are doing virtually the same thing.
Can we keep only one?

Pay attention to the `desc || id` logic :)